File: app/src/main/java/com/kakacat/minitool/bingpic/ImageAdapter.java
package com.kakacat.minitool.bingpic;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.facebook.drawee.view.SimpleDraweeView;
import com.kakacat.minitool.R;
import com.kakacat.minitool.common.myinterface.RecycleViewItemOnClickListener;
import com.kakacat.minitool.common.myinterface.RecycleViewItemOnLongClickListener;
import com.kakacat.minitool.common.myinterface.RecycleViewOnTouchListener;

import java.util.List;

public class ImageAdapter extends RecyclerView.Adapter<ImageAdapter.ViewHolder> {

    private List<String> addressList;
    private LayoutInflater inflater;

    private RecycleViewItemOnClickListener onClickListener;
    private RecycleViewItemOnLongClickListener onLongClickListener;
    private RecycleViewOnTouchListener onTouchListener;

    public ImageAdapter(List<String> addressList) {
        this.addressList = addressList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        if (inflater == null) {
            inflater = LayoutInflater.from(parent.getContext());
        }
        View view = inflater.inflate(R.layout.bing_pic_item_layout, parent, false);
        return new ViewHolder(view);
    }

    @SuppressLint("ClickableViewAccessibility")
    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.sdv.setImageURI(addressList.get(position));

        if (onClickListener != null)
            holder.sdv.setOnClickListener(v -> onClickListener.onClick(holder.itemView, position));
        if (onLongClickListener != null) {
            holder.sdv.setOnLongClickListener(v -> {
                onLongClickListener.onLongClick(holder.sdv, position);
                return false;
            });
        }
        if (onTouchListener != null) {
            holder.sdv.setOnTouchListener((v, event) -> {
                onTouchListener.onTouch(v, event);
                return false;
            });
        }
    }

    @Override
    public int getItemCount() {
        return addressList.size();
    }


    public void setOnClickListener(RecycleViewItemOnClickListener onClickListener) {
        this.onClickListener = onClickListener;
    }

    public void setOnLongClickListener(RecycleViewItemOnLongClickListener onLongClickListener) {
        this.onLongClickListener = onLongClickListener;
    }

    public void setOnTouchListener(RecycleViewOnTouchListener onTouchListener) {
        this.onTouchListener = onTouchListener;
    }


    static class ViewHolder extends RecyclerView.ViewHolder {

        private SimpleDraweeView sdv;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            sdv = itemView.findViewById(R.id.image_view);
        }
    }
}

File: app/src/main/java/com/kakacat/minitool/bingpic/ImageAdapter.kt
package com.kakacat.minitool.bingpic

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.facebook.drawee.view.SimpleDraweeView
import com.kakacat.minitool.R
import com.kakacat.minitool.common.ui.RecycleViewListener


class ImageAdapter(private val addressList: List<String?>) : RecyclerView.Adapter<ImageAdapter.ViewHolder>() {

    private var inflater : LayoutInflater? = null
    private var onItemClickListener: RecycleViewListener.OnItemClick? = null
    private var onLongClickListener: RecycleViewListener.OnItemLongClick? = null
    private var onTouchListener: RecycleViewListener.OnTouch? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        if(inflater == null){
            inflater = LayoutInflater.from(parent.context)
        }
        return ViewHolder(inflater!!.inflate(R.layout.bing_pic_item_layout,parent,false))
    }

    override fun getItemCount() = addressList.size

    @SuppressLint("ClickableViewAccessibility")
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.sdv!!.setImageURI(addressList[position])
        if(onItemClickListener != null){
            holder.sdv!!.setOnClickListener{ onItemClickListener!!.onClick(holder.itemView,position)}
        }
        if(onLongClickListener != null){
            holder.sdv!!.setOnLongClickListener{
                onLongClickListener!!.onLongClick(holder.sdv,position)
                false
            }
        }
        if(onTouchListener != null){
            holder.sdv!!.setOnTouchListener { v, event ->
                onTouchListener!!.onTouch(v,event)
                false
            }
        }
    }

    fun setOnClickListener(listener: RecycleViewListener.OnItemClick){
        this.onItemClickListener = listener
    }

    fun setOnLongClickListener(listener: RecycleViewListener.OnItemLongClick) {
        this.onLongClickListener = listener
    }

    fun setOnTouchListener(listener: RecycleViewListener.OnTouch) {
        this.onTouchListener = listener
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){

        var sdv : SimpleDraweeView? = null

        init {
            this.sdv = itemView.findViewById(R.id.image_view)
        }
    }
}